### PR TITLE
fix(codegen/nodejs): generate the required method param documentation correctly

### LINKED ISSUE #4132: Fix nodejs codegen for required param documentation

### PR DESCRIPTION
**Pull request by dolanor**
Fixes #4132

### CODE REVIEW

File: sdk/nodejs/api/client.gen.ts [hunk 1]
```diff
@@ -289,6 +291,7 @@ export class Container extends BaseClient {
 
   /**
    * Retrieve a directory at the given path. Mounts are included.
+   *
```
Comment (slumbering) [inline sdk/nodejs/api/client.gen.ts:294]: The codegen is generating unnecessary extra line. Comment valid for the whole file.

File: sdk/nodejs/api/client.gen.ts [hunk 2]
```diff
@@ -252,6 +252,8 @@ export class CacheVolume extends BaseClient {
 export class Container extends BaseClient {
   /**
    * Initialize this container from a Dockerfile build
+   *
+   * @param opts optional params for build
```
Comment (slumbering) [inline sdk/nodejs/api/client.gen.ts:256]: Is there a real benefit in terms of DX to add this @param ? Do you think it could be possible to display an exhaustive list of those opts params ?

Comment (dolanor) [reply sdk/nodejs/api/client.gen.ts:256]: I changed my opinion about it. It is not standard, but it would help.
I suggest: `@param opts.<field> ipsem lorem`
I'll add it

File: sdk/nodejs/api/client.gen.ts (hunk 1 above)
Comment (dolanor) [reply sdk/nodejs/api/client.gen.ts:294]: We decided to remove any extra line, to be idiomatic with jsdoc and avoid this bikeshedding again: https://jsdoc.app/about-getting-started.html

File: sdk/nodejs/api/client.gen.ts (hunk 2 above)
Comment (TomChv) [reply sdk/nodejs/api/client.gen.ts:256]: Those decorators are only useful if you generate doc from source code using [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) for example.
I do not find it useful there since the field is already named. 
I think you can stick either to `opts.<field> lorem` or just `<fields> lorem`.

Comment (slumbering) [reply sdk/nodejs/api/client.gen.ts:256]: > Those decorators are only useful if you generate doc from source code using [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) for example. I do not find it useful there since the field is already named. I think you can stick either to `opts.<field> lorem` or just `<fields> lorem`.

But those decorators are also used by typedoc : https://typedoc.org/tags/param/#object-literals
Even if it works without it, I think we semantically need them, does it make sense ?

Comment (TomChv) [reply sdk/nodejs/api/client.gen.ts:256]: You're right!

**Comment by slumbering** [comment]
@dolanor did you have the chance to tackle this one ? Otherwise, do you think it can be done this week ? Thanks

**Comment by dolanor** [comment]
@slumbering : I've fixed all the comments we had. You can check it out.

File: codegen/generator/nodejs/templates/src/method_comment.ts.tmpl [hunk 3]
```diff
@@ -1,32 +1,41 @@
...
+	{{- $argsDesc := ArgsHaveDescription .Args }}
+	{{- $deprecationLines := FormatDeprecation .DeprecationReason }}
+	{{- if or .Description $argsDesc .IsDeprecated }}
+		{{- $commentLines := CommentToLines .Description }}
+		{{- /* we split the comment string into a string slice of one line per element */ -}}
 {{""}}
 {{""}}
   /**
-  {{- range $commentLines }}
+		{{- range $commentLines }}
```
Comment (TomChv) [inline codegen/generator/nodejs/templates/src/method_comment.ts.tmpl:12]: Nit: Why so many tab? 

:bulb: It's definitely not a merge blocker, we can take care of this on #4294

**Review by TomChv — APPROVED** [review]
I love the changes on this PR. :rocket:

Params look good this way and tt makes so much sense to compare files instead of plain string!

File: codegen/generator/nodejs/templates/src/method_comment.ts.tmpl (hunk 3 above)
Comment (dolanor) [reply codegen/generator/nodejs/templates/src/method_comment.ts.tmpl:12]: So, this is just to be able to view the logic part of the template compared to the actual formatting (which needs to be 2-space indented, etc)
Otherwise, if all the logic is at the same level of indentation, it's hard to follow anything.

File: sdk/nodejs/api/client.gen.ts [hunk 4]
```diff
@@ -382,9 +382,13 @@ export class Container extends BaseClient {
 
   /**
    * This container after executing the specified command inside it
-   *
-   * @param opts optional params for exec
-   *
+   * @param opts.args Command to run instead of the container's default command
```
Comment (slumbering) [inline sdk/nodejs/api/client.gen.ts:385]: This is awesome to have explicit opts args 👍 !!
When I look into `client.gen.ts`, those explicit params are only for `exec`, `withExec`. 
Is this something that we can generalize to every method that takes opts ? For instance:
```ts
build(
    context: DirectoryID | Directory,
    opts?: ContainerBuildOpts
  )
  ```
This can be done in a follow up PR, this is not a blocker as this PR is getting old and should be merged.

Comment (dolanor) [reply sdk/nodejs/api/client.gen.ts:385]: There is a good reason for this: There is no documentation on every other params for functions in the GraphQL documentation. Hence: https://github.com/dagger/dagger/issues/4129 and https://github.com/dagger/dagger/pull/4240
Good thing, adding doc in the schema will add doc in our code.

Comment (slumbering) [reply sdk/nodejs/api/client.gen.ts:385]: Make sense. Thanks for the clarification
